### ISSUE TITLE
Speed up CI

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [RagnarGrootKoerkamp/BAPCtools#161]
Ideally <1min

**Comment by mpsijm** [comment]
It seems like running TLE submissions in `test_problems.py` is also a large part of the time spent in tests.

**Comment by mpsijm** [comment]
Just had an idea: maybe we can "mock" LaTeX compilation by inserting a shell script onto the `PATH` in the test setup, that replaces the `latexmk` binary and only copies its input to a `.pdf` file. At the very least, we can do this for any test that creates a ZIP. We can then leave the actual LaTeX compilation in tests like `test_pdf`, `test_solutions`, `test_problem_slides`, and `test_new_contest_problem`, because it's probably still useful to verify that the template LaTeX files compile correctly as well.

I do still think that zipping the problems is a useful test, especially because of https://github.com/RagnarGrootKoerkamp/BAPCtools/pull/441#issuecomment-2764494385.

**Comment by RagnarGrootKoerkamp** [comment]
Right now this got much much worse because the WSL CI installs a bunch of stuff on every run. Really we should avoid that and either use a cache or a separate docker image or so.

**Comment by mpsijm** [comment]
Yeah, that's a whole separate issue 😛 If I want to keep moving, I simply ignore the result of the WSL tests and merge anyway. If they turn out to fail later, we can always fix that after merging, that shouldn't happen too often anyway :joy: Nonetheless, that's indeed something we want to improve at some point, and your suggestions sound good 🙂
